src/main/java/at/hannibal2/skyhanni/features/combat/end/ProfitPerDragon.kt

```diff
@@ -28,9 +28,13 @@ object ProfitPerDragon {
     private fun scanForLoot() {
         val entities = EntityUtils.getEntities<EntityArmorStand>()
 
-        for (uuid in scannedLootUUIDs) {
+        val scannedLootIterator = scannedLootUUIDs.iterator()
+
+        while (scannedLootIterator.hasNext()) {
+            val uuid = scannedLootIterator.next()
+
             if (entities.none { it.uniqueID == uuid }) {
-                scannedLootUUIDs.remove(uuid)
+                scannedLootIterator.remove()
```

Comment:
```suggestion
                scannedLootIterator.remove(uuid)
```
I presume you meant this?

Reply:
> I presume you meant this?

```
Removes from the underlying collection the last element returned by this iterator.
```

The iterator remove function has this definition so the remove by itself will remove the current element that we are on form the list.